src/main/java/com/w6/nlp/Parser.java

```diff
@@ -59,25 +69,41 @@ public Response generateResponse(final String input) {
             List<String> sentenseWhere = new ArrayList<String>();
             List<String> sentenseWhen = new ArrayList<String>();
             List<String> sentenseWhat = new ArrayList<String>();
+            
+            
+            
+            
             Tree parse = lp.apply(
                     tokenizerFactory.getTokenizer(new StringReader(sentence.text()))
                         .tokenize()
             );
+            
+            sentenseWhat = violentVerbsParser.getAllViolentVerbs(parse);
+            
             sentenseWhen = DateTimeParser.parseDateAndTimeFromString(
                 sentence, 
                 dateTimeTags
             );
+            
             sentenseWhere = DateTimeParser.parseDateAndTimeFromString(
                 sentence, 
                 locationTags
             );
-            sentenseWhat = violentVerbsParser.getAllViolentVerbs(parse);
-        
+            
+            int typeOfSentence = 1;
             sentenseWeapon = weaponsParser.getAllWeapons(parse);
-            ObjectsAndSubjects objAndSubj = GetDoerAndVictim.getSubjectAndObjectOfViolence(parse,sentenseWhat);
-
-            sentenseWho.addAll(objAndSubj.subjects);
-            sentenseWhom.addAll(objAndSubj.objects);
+            
+            if (!sentenseWhat.isEmpty())
+            {
+                ObjectsAndSubjects objAndSubj = GetDoerAndVictim.getSubjectAndObjectOfViolence(parse,sentenseWhat);
+                sentenseWho.addAll(objAndSubj.subjects);
+                sentenseWhom.addAll(objAndSubj.objects);
+                typeOfSentence = 2;
+            }  
+            
+            addValueToRatedArray(typeOfSentence, ratedWhere, sentenseWhere);
+            addValueToRatedArray(typeOfSentence, ratedWhen, sentenseWhen);
```

Comment:
I think logic with type of sentence should be for every entry. Location in violent sentence is more important that location in other sentence.